mkimage

```diff
@@ -7,6 +7,13 @@ ROOT=$(cd "$(dirname "$0")" && pwd)
 
 TARGET=${1:?Specify the target filename}
 DIST=${2:-stable}
+SNAPSHOT_ID=${3:-}
+
+# TRIM -snapshot from the distro
+IFS="-"
+read -ra DISTARR <<< "$DIST"
+DIST="${DISTARR[0]}"
+unset IFS
```

Comment:
I don't like the "IFS" solution. I would use:
```
read -r -a DISTARR <<< "$(tr '-' ' ' <<< "$DIST")"
```

Reply:
meta: in this case it's not just a matter of personal style/taste, so it might be worth avoid expressing the review comment as such. 
Example:

"The IFS approach is brittle since it's a global setting and there is a risk of forgetting to reset it to the previous state. For example, you overwrite IFS="-" assuming that it was unset before"...

Reply:
anyway, a much simpler solution is:

```
DIST=${DIST%%-*}
```

Reply:
I think we could add this better

```diff
- DIST=${2:-stable}
+ DIST=${2%%-*}
```

since `$2` is not going to never be empty, right?

Reply:
sorry, I'm not really reviewing the logic of this PR, my suggestion is just a bash TIP:

if you have a string `foo-bar-baz` and you want to get `foo`, then `FOO=${FOO%%-*}` will do the trick.

Reply:
and `FOO=${FOO%-*}` instead would just strip the last one, i.e. it will yield `foo-bar`

Reply:
w.r.t 2 being empty or not, another general suggestion: if you're making assumption about input invariants, please add an assertion that will crash the script if it's violated.

Reply:
I think in this case it's simpler, the parameter doesn't requiere to fit any assertion but given a name if it has the suffix `-snapshot` then we will use the latest snapshot id in debian repositories. e.g:

```bash
# if the script receive buster-snapshot it will create 
minideb:buster-snapshot-<latest_id>
# using https://snapshot.debian.org/archive/debian/<latest_id>/

# if the script receive my-buster-snapshot it will create
minideb:my-buster-snapshot-<latest_id>
# using https://snapshot.debian.org/archive/debian/<latest_id>/

# if the script receive my-buster or buster it will create
minideb:my-buster
minideb:buster
# using http://deb.debian.org/debian
```

I mean, you should not have to fit any requirement, you could choose to make a snapshot version of the distro.

Reply:
done!